main.py

```diff
@@ -1,19 +1,23 @@
-import sys
 import argparse
-import networkx as nx
-from productions.p1 import P1
+import sys
 import matplotlib.pyplot as plt
+import networkx as nx
+from productions import P1, P2
+from visualization import draw_graph
+
 
 def main(args):
     G = nx.Graph()
-    G.add_node(1, node_color='red', label='El', pos=(0, 0))
+    G.add_node(1, label='El', pos=(0, 0), layer=0)
 
     P1.apply(G)
+    P2.apply(G)
+    P2.apply(G)
+    P2.apply(G)
 
-    nx.draw(G, nx.get_node_attributes(G,'pos'), labels=nx.get_node_attributes(G,'label'), font_size=10)
-    plt.show()
+    draw_graph(G)
+    # draw_graph(G, 1) # draw layer 1
```

Comment:
```suggestion
```

Reply:
I would leave this comment for now just to give people a hint that this is posible without delving into implementation. Later this could be removed.

Reply:
outdated already, resolving